src/common/resolver.rs

```diff
@@ -100,6 +101,13 @@ impl Resolver {
         })
     }
 
+    pub fn parse_name(&self, key: &str) -> crate::Result<Name> {
```

Comment:
No need to create a function for this, instead use:

```rust
Name::from_str_relaxed(key).map_err(|err| Error::DnsError(error.to_string()))?
```

Reply:
Actually, a `From<Error>` implementation is better rather than mapping each time.